lib/rules/define-macros-order.js

```diff
@@ -0,0 +1,255 @@
+            return
+          }
+
+          // need move both defineEmits and defineProps
+          if (firstStatement !== shouldFirstNode) {
+            reportBothNotOnTop(
+              shouldFirstNode,
+              shouldSecondNode,
+              firstStatement
+            )
+            return
+          }
+
+          // need move only second
+          if (secondStatement !== shouldSecondNode) {
+            reportNotOnTop(order[1], shouldSecondNode, shouldFirstNode)
+          }
+
+          return
+        }
+
+        // have only first and need to move it
+        if (shouldFirstNode && firstStatement !== shouldFirstNode) {
+          reportNotOnTop(order[0], shouldFirstNode, firstStatement)
+          return
+        }
+
+        // have only second and need to move it
+        if (shouldSecondNode && firstStatement !== shouldSecondNode) {
+          reportNotOnTop(order[1], shouldSecondNode, firstStatement)
+        }
+      }
+    }
+  )
+
+  /**
+   * @param {ASTNode} shouldFirstNode
+   * @param {ASTNode} shouldSecondNode
+   * @param {ASTNode} before
+   */
+  function reportBothNotOnTop(shouldFirstNode, shouldSecondNode, before) {
+    context.report({
+      node: shouldFirstNode,
+      loc: shouldFirstNode.loc,
+      messageId: 'macrosNotOnTop',
+      data: {
+        macro: order[0]
+      },
+      fix(fixer) {
+        return [
+          ...moveNodeBefore(fixer, shouldFirstNode, before),
+          ...moveNodeBefore(fixer, shouldSecondNode, before)
+        ]
+      }
+    })
+  }
+
+  /**
+   * @param {string} macro
+   * @param {ASTNode} node
+   * @param {ASTNode} before
+   */
+  function reportNotOnTop(macro, node, before) {
+    context.report({
+      node,
+      loc: node.loc,
+      messageId: 'macrosNotOnTop',
+      data: {
+        macro
+      },
+      fix(fixer) {
+        return moveNodeBefore(fixer, node, before)
+      }
+    })
+  }
+
+  /**
+   * Move one newline with "node" to before the "beforeNode"
+   * @param {RuleFixer} fixer
+   * @param {ASTNode} node
+   * @param {ASTNode} beforeNode
+   */
+  function moveNodeBefore(fixer, node, beforeNode) {
+    const beforeNodeToken = sourceCode.getTokenBefore(node, {
+      includeComments: true
+    })
+    const beforeNodeIndex = getNewLineIndex(node)
+    const textNode = sourceCode.getText(node, node.range[0] - beforeNodeIndex)
+    /** @type {[number, number]} */
+    const removeRange = [beforeNodeToken.range[1], node.range[1]]
+    const index = getNewLineIndex(beforeNode)
+
+    return [
+      fixer.insertTextAfterRange([index, index], textNode),
+      fixer.removeRange(removeRange)
+    ]
+  }
+
+  /**
+   * Get index of first new line before the "node"
+   * @param {ASTNode} node
+   * @return {number}
+   */
+  function getNewLineIndex(node) {
+    const after = sourceCode.getTokenBefore(node, { includeComments: true })
+    const hasWhitespace = node.loc.start.line - after.loc.end.line > 1
```

Comment:
```suggestion
    const hasWhitespace = after.loc.end.line < node.loc.start.line
```

Reply:
I removed this heuristic about new lines. Here are the details: https://github.com/vuejs/eslint-plugin-vue/pull/1856#discussion_r854017385